### ISSUE TITLE
Cannot connect to AS4 backing store from docker container

### ISSUE THREAD

**Issue by yxuco** [TIBCOSoftware/be-tools#299]
### Version Information
| Software       | Version(s)     |
| -------------- | -------------- |
| BusinessEvents | 6.2.2 HF001               |
| OS Type?       | MacOS               |
| OS Version?    |                |
| Docker         | Docker Desktop 4.12.0 (85629)               |
| Kubernetes     |                |
| Helm           |                |

### What is the expected behavior?
Docker container should connect to AS4 backing store on the host machine, since the project works when running locally.
### What is the actual behavior?
Following is the error:
```
2022 Oct 16 17:40:05:676 GMT Z be-engine INFO [main] - [StoreProviderFactory] Using store provider - 'com.tibco.be.as.store.ActiveSpacesStoreProvider'
2022 Oct 16 17:40:08:751 GMT Z be-engine FATAL [main] - [as.store] ActiveSpaces BackingStore initialization failed.
java.lang.RuntimeException: Error occurred while connecting to Store with url[http://host.docker.internal:8080]
	at com.tibco.cep.store.functions.StoreFunctions.connect(StoreFunctions.java:80)
	at com.tibco.be.as.store.ASStoreConnectionManager.connectASStore(ASStoreConnectionManager.java:61)
	at com.tibco.be.as.store.ASStoreConnectionManager.init(ASStoreConnectionManager.java:45)
	at com.tibco.be.as.store.ActiveSpacesStore.initialize(ActiveSpacesStore.java:150)
	at com.tibco.be.as.store.ActiveSpacesStore.<init>(ActiveSpacesStore.java:102)
	at com.tibco.be.as.store.ActiveSpacesStoreProvider.<init>(ActiveSpacesStoreProvider.java:72)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.tibco.cep.runtime.service.store.StoreProviderFactory.getStoreProvider(StoreProviderFactory.java:40)
	at com.tibco.cep.runtime.service.cluster.MultiAgentCluster.<init>(MultiAgentCluster.java:103)
	at com.tibco.cep.runtime.service.cluster.CacheClusterProvider.createCluster(CacheClusterProvider.java:76)
	at com.tibco.cep.runtime.session.impl.RuleServiceProviderImpl.initCluster(RuleServiceProviderImpl.java:803)
	at com.tibco.cep.runtime.session.impl.RuleServiceProviderImpl.configure(RuleServiceProviderImpl.java:299)
	at com.tibco.cep.container.standalone.BEMain.jumpStart(SourceFile:169)
	at com.tibco.cep.container.standalone.BEMain.main(SourceFile:71)
Caused by: com.tibco.datagrid.exceptions.DataGridTimeoutException: Response not received within timeout interval specified.
	at com.tibco.datagrid.jni.NativeDataGrid.createNativeConnection(Native Method)
	at com.tibco.datagrid.jni.NativeDataGrid.connect(NativeDataGrid.java:107)
	at com.tibco.datagrid.DataGrid.connect(DataGrid.java:91)
	at com.tibco.cep.store.as.ASConnection.connect(ASConnection.java:93)
	at com.tibco.cep.store.functions.StoreFunctions.connect(StoreFunctions.java:78)
	... 16 more
Caused by: com.tibco.datagrid.DataGridException: TIBCO Exception:
	Error Code    = Timeout
	Description   = Response not received within timeout interval specified.
	Thread Name   = be-engine
	Stack Trace:
		_tibdgRequestor_GetReplies, 972
		_tibdgConnection_Bind, 1264
		_tibdgConnection_Create, 1863
		_tibdgGrid_Connect, 2450
		tibdgGrid_Connect, 2489

	... 21 more
```
### Please provide a unit test that demonstrates the bug.
On Macbook pro, build the attached BE project, and build docker image using the ear and fdstore.cdd, which uses FTL cluster + AS4 backing store.
```
./build_image.sh -i app -s $LINUX_INSTALLERS -a $BUILD_DIR -t fdapp:v1.0.0 --gv-provider "consul,http"
```
where linux installers folder contains zips for BE 6.2.2, HF-001, FTL 6.8.1, AS 4.8.0.

Start the docker container using
```
docker run --rm --name inf1 -e FTL_gv_REALM_SERVER=http://host.docker.internal:8585 -e AS_gv_REALM_SERVER=http://host.docker.internal:8080 -e HTTP_gv_PORT=8209 -e PU=inf1 -p 8209:8209 fdapp:v1.0.0
```

I have already started FTL server on Mac localhost on port 8585, and AS4 on port 8080, and setup the AS tables for this app, and confirmed the FTL and AS setup by running the same app on localhost without docker.

After starting the docker container, it connected to the FTL cluster (port 8585) successfully with the following log, but failed to connect to the AS backing store.

```
2022 Oct 16 17:40:05:611 GMT Z be-engine INFO [main] - [ftl.cluster] FTL Server [http://host.docker.internal:8585] is available.
2022 Oct 16 17:40:05:611 GMT Z be-engine DEBUG [main] - [ftl.cluster] Execute GET on URL(http://host.docker.internal:8585/api/v1/realm/persistence/ftl.default.cluster)
2022 Oct 16 17:40:05:616 GMT Z be-engine DEBUG [main] - [ftl.cluster] HTTP/1.1 200 OK
2022 Oct 16 17:40:05:618 GMT Z be-engine INFO [main] - [ftl.cluster] Cluster [ftl.default.cluster] exist on FTL Server [http://host.docker.internal:8585]
2022 Oct 16 17:40:05:618 GMT Z be-engine DEBUG [main] - [ftl.cluster] Execute GET on URL(http://host.docker.internal:8585/api/v1/realm/persistence/ftl.default.cluster/stores/cep.ftl.persistent.store-ftl.default.cluster$fdstore)
2022 Oct 16 17:40:05:625 GMT Z be-engine DEBUG [main] - [ftl.cluster] HTTP/1.1 200 OK
2022 Oct 16 17:40:05:625 GMT Z be-engine INFO [main] - [ftl.cluster] Persistent Store [cep.ftl.persistent.store-ftl.default.cluster$fdstore] exist on FTL Server [http://host.docker.internal:8585]
2022 Oct 16 17:40:05:626 GMT Z be-engine DEBUG [main] - [ftl.cluster] Execute GET on URL(http://host.docker.internal:8585/api/v1/realm/applications/ftl.default.cluster$fdstore)
2022 Oct 16 17:40:05:628 GMT Z be-engine DEBUG [main] - [ftl.cluster] HTTP/1.1 200 OK
2022 Oct 16 17:40:05:629 GMT Z be-engine INFO [main] - [ftl.cluster] Application [ftl.default.cluster$fdstore] exist on FTL Server [http://host.docker.internal:8585]
2022 Oct 16 17:40:05:629 GMT Z be-engine INFO [main] - [ftl.cluster] CEP Cluster/Application configuration ftl.default.cluster/ftl.default.cluster$fdstore exist on FTL Server [http://host.docker.internal:8585]
2022 Oct 16 17:40:05:653 GMT Z be-engine INFO [main] - [ftl.cluster] Connected to FTL server [http://host.docker.internal:8585] using [TIBCO FTL Version 6.8.1   V3]
```

### If the bug is confirmed, would you be willing to submit a PR?
Yes / No _(Help can be provided if you need assistance submitting a PR)_
[FraudDetectionStore.zip](https://github.com/TIBCOSoftware/be-tools/files/9796676/FraudDetectionStore.zip)

**Comment by rameshpolishetti** [comment]
@yxuco I have followed steps as mentioned in BE shipped example `BE_HOME/be/6.2/examples/standard/FraudDetectionStore/readme.html`. Able to connect to AS4 successfully.
Both FTL & AS4 started on my Mac localhost on port 8585 (FTL) and 5055(as4). BE application started on docker with be below command:
`docker run -e FTL_gv_REALM_SERVER=http://host.docker.internal:8585 -e AS_gv_REALM_SERVER=http://host.docker.internal:5055 -e PU=default fdc:ftl`

Attached logs for reference.


Debug note: Please check whether you have changed anything in the FraudDetectionStore/script - (AS4 related configuration).
My configuration contains localhost as AS4 hostname:
`
HOSTPORT=localhost:5055
`



[inf.txt](https://github.com/TIBCOSoftware/be-tools/files/9803270/inf.txt)

**Comment by yxuco** [comment]
Yes, my project is different from the standard sample for fdstore.  However, I found that the issue is the use of AS 4.8.0, which does not seem to work when AS client is in a docker container.  After I downgrade to AS 4.7.1, everything is working as expected.  So please close this issue.
